lib/adapters/QUnitAdapter.js

```diff
@@ -12,22 +12,50 @@ export default class QUnitAdapter extends EventEmitter {
     QUnit.testDone(this.onTestDone.bind(this))
     QUnit.moduleDone(this.onModuleDone.bind(this))
     QUnit.done(this.onDone.bind(this))
-
-    this.idDataMap = {}
   }
 
   convertModule (qunitModule) {
     var suiteName = qunitModule.name
     var tests = []
+
     for (let qunitTest of qunitModule.tests) {
-      let test = new Test(qunitTest.name, suiteName)
-      tests.push(test)
-      this.idDataMap[qunitTest.testId] = test
+      tests.push(this.convertTest(qunitTest, suiteName))
+    }
+
+    return new Suite(suiteName, [], tests)
+  }
+
+  /**
+   * @param qunitTest - when passed in from the convertModule function it does
+   *  contain only its "name" and the "testId".
+   * @param suiteNameParam - is used only when calling convertTest from
+   *  the convertModule function, in rest it is undefined.
+   */
+  convertTest (qunitTest, suiteNameParam) {
```

Comment:
`suiteNameParam` is a weird parameter, since its used only once. I suggest renaming this to `suiteName`, then overwriting this as necessary.